io.openems.edge.evcs.goe.chargerhome/src/io/openems/edge/evcs/goe/chargerhome/GoeChargerHomeImpl.java

```diff
@@ -0,0 +1,340 @@
+				this.channel(Evcs.ChannelId.CHARGE_POWER).setNextValue(nrg.get(11).getAsInt()*10);
+				this.channel(Evcs.ChannelId.CHARGING_TYPE).setNextValue(ChargingType.AC);
+				
+				this.channel(GoeChannelId.ENERGY_TOTAL).setNextValue(json.get("eto").getAsInt()*100);				
+				this.channel(Evcs.ChannelId.ENERGY_SESSION).setNextValue(json.get("dws").getAsInt()*10/3600);
+			
+				this.channel(GoeChannelId.ERROR).setNextValue(err);
+				this.channel(Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED).setNextValue(false);
+				
+				this.setEnergySession();
+				this.setPower();				
+				
+			}
+			else{
+				this.channel(Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED).setNextValue(true);
+			}
+			
+			break;
+		}
+	}
+	
+	private int ConvertGoeStatus(int status) {
+		switch(status) {
+		case 1: // ready for charging, car unplugged
+			return Status.NOT_READY_FOR_CHARGING.getValue();
+		case 2: // charging
+			return Status.CHARGING.getValue();
+		case 3: // waiting for car
+			return Status.READY_FOR_CHARGING.getValue(); 
+		case 4: // charging finished, car plugged
+			return Status.CHARGING_FINISHED.getValue();
+		default:
+			return Status.UNDEFINED.getValue();
+		}
+	}
+	
+	private int ConvertGoePhase(int phase) {
+		int Phasen = (byte)phase & 0b00111000;			
+		switch(Phasen) {
+		case 8: // 0b00001000: Phase 1 is active
+			return 1;
+		case 24: // 0b00011000: Phase 1+2 is active
+			return 2;
+		case 56: // 0b00111000: Phase1-3 are active
+			return 3;
+		default:
+			return 0;
+		}
+	}
+	
+	private int GetMaxHardwarePower(int cableCurrent) {
+		int MaxPower = 0;
+		int phases = 3;
+		if (cableCurrent < this.MaxCurrent && cableCurrent > 0) {
+			MaxPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MaxPower = phases*this.MaxCurrent*230/1000;
+		}
+		return MaxPower;
+	}
+	
+	private int GetMinHardwarePower(int phases, int cableCurrent) {
+		int MinPower = 0;
+		int MinPowerHw = phases*this.MinCurrent*230/1000;
+		if (cableCurrent < this.MinCurrent && cableCurrent > 0) {
+			MinPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MinPower = phases*this.MinCurrent*230/1000;
+		}
+		if (MinPower > MinPowerHw || MinPower < 0) {
+			return MinPowerHw;
+		}
+		else {
+			return MinPower;
+		}
+	}
+		
+	private int GetMaxPower(int phases) {
+		return phases*this.MaxCurrent*230/1000;
+	}
+	
+	/**
+	 * Sets the current from SET_CHARGE_POWER channel.
+	 * 
+	 * <p>
+	 * Allowed loading current are between MinCurrent and MaxCurrent. Invalid values are
+	 * discarded. The value is also depending on the DIP-switch settings and the
+	 * used cable of the charging station.
+	 */
+	private void setPower() {
+		
+		WriteChannel<Integer> energyLimitChannel = this.channel(ManagedEvcs.ChannelId.SET_ENERGY_LIMIT);
+		int energyLimit = energyLimitChannel.getNextValue().orElse(0);
+
+		// Check energy limit
+		if (energyLimit == 0 || energyLimit > this.getEnergySession().orElse(0)) {
```

Comment:
A Option to set "Charge until Battery has xy%" would also be nice

Reply:
Yes of course, but this should be added in the nature and used in the controller.